HIP/hip-906.md

```diff
@@ -0,0 +1,125 @@
+requested-by: EVM developers
+type: Standards Track 
+category: Service
+needs-council-approval: Yes
+status: Review
+created: 2024-02-23
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/906
+updated: 2024-03-06
+requires: 632
+---
+
+## Abstract
+
+The Smart Contract service currently provides functionality to grant allowance and approval for tokens from an owner account to a spender account. 
+However, there is currently no way to grant allowance and approval for hbars from an owner account using the Smart Contract service. This HIP proposes to remedy this omission.
+
+## Motivation
+
+Smart Contracts developers face an obstacle when implementing certain use cases for transferring hbars between accounts and contracts. 
+The lack of a mechanism to grant allowance and approval for hbars without using HAPI hinders developers’ workflow. Providing this functionality will enhance the developer experience and remove this limitation.
+
+## Rationale
+
+This HIP proposes adding the missing functionality via a new interface called `IHRC632`, which will act on an account address. 
+
+An alternative approach would be to introduce a new interface that a contract can implement. However, this would require deploying a contract, which may not always be desired. 
+Additionally, taking this approach could potentially violate the smart contracts security model, as the sender of the frame when executing the system contract would be the deployed contract rather than the EOA (Externally Owned Account), which would not be desirable in most cases.
+
+
+## User stories
+
+1. As a smart contract developer, I want to be able to grant an allowance for hbars from an EOA to a spender account or contract.
+2. As a smart contract developer, I want to be able to grant an allowance for hbars from a contract to a spender account or contract.
+3. As a smart contract developer, I want to be able to get the current allowance of hbars from an owner account to a spender account or contract.
+
+## Specification
+
+HIP-632 proposes to introduce a new system contract for accessing Hedera account functionality (Hedera Account Service - HAS).
+This HIP extends the functionality of the new system contract by adding another related interface `IHRC632` to support the `hbarAllowance` and `hbarApprove` functions.
+The `hbarAllowance` function will be used to retrieve information about allowance granted to a spender and the `hbarApprove` function will allow the sender to grant to the `spender` an allowance of hbars.
+
+
+Similar to the way redirection to a proxy contract during EVM execution for tokens works [see HIP-719](https://github.com/hashgraph/hedera-improvement-proposal/blob/main/HIP/hip-719.md),
+this HIP proposes to introduce a new proxy contract for accounts.  During EVM execution, if the target of the current frame is an account, a call to the new proxy contract will be created and the current calldata will be injected into 
```

Comment:
Are we introducing a new proxy contract or using the same?
If the same then I guess what we're introducing is a new proxy pathway for accounts

Reply:
I think it is best to introduce a new proxy contract which is very similar to the existing token proxy contract except for the contract address (0x16a).  This way we are clearly delineating that we are acting on an account vs tokens.  

This argument could break down if we want to treat hbars as a **special** token but this is not an avenue we want to explore at this time.

Reply:
Great, wanted to make sure new proxy was clear